rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -536,6 +554,112 @@ void saveOldFederation_after_RSKIP123_should_save_null() {
         );
     }
 
+    private static Stream<Arguments> provideFederationBtcUTXOsTestArguments() {
+        return Stream.of(
+            Arguments.of(false, false, NetworkParameters.ID_TESTNET),
+            Arguments.of(false, false, NetworkParameters.ID_MAINNET),
+            Arguments.of(false, true, NetworkParameters.ID_TESTNET),
+            Arguments.of(false, true, NetworkParameters.ID_MAINNET),
+            Arguments.of(true, false, NetworkParameters.ID_TESTNET),
+            Arguments.of(true, false, NetworkParameters.ID_MAINNET),
+            Arguments.of(true, true, NetworkParameters.ID_TESTNET),
+            Arguments.of(true, true, NetworkParameters.ID_MAINNET)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideFederationBtcUTXOsTestArguments")
+    void testGetNewFederationBtcUTXOsWithCombinationsOfRSKIPsAndNetworks(boolean isRskip284Active, boolean isRskip293Active, String networkId) {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(isRskip284Active);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(isRskip293Active);
+
+        NetworkParameters networkParameters = NetworkParameters.fromID(networkId);
+
+        Address btcAddress = new Address(testnetBtcParams, Hex.decode("4a22c3c4cbb31e4d03b15550636762bda0baf85a"));
+
+        Repository repository = mock(Repository.class);
+        List<UTXO> federationUtxos = BitcoinTestUtils.createUTXOs(1, btcAddress);
+        when(repository.getStorageBytes(
+            bridgeAddress,
+            NEW_FEDERATION_BTC_UTXOS_KEY.getKey()
+        )).thenReturn(BridgeSerializationUtils.serializeUTXOList(federationUtxos));
+
+        List<UTXO> federationUtxosAfterRskip284Activation = BitcoinTestUtils.createUTXOs(2, btcAddress);
```

Comment:
Is it ok to assume RSKIP 284 is active? The test receives a boolean indicating it could or could not be active

Reply:
Those are the expected utxos for when those rskips are active. I could refactor this test and separate it into multiple tests, one for ech rskip. What do you think?

Reply:
yes, i find this flow a bit confusing. I think its better if you split the tests between rskips and networks

Reply:
you could use Nested to reuse the setup

Reply:
Or maybe do this only when RSKIP284 is active.

Could be after the part `if (!activations.isActive(ConsensusRule.RSKIP284)) {`

Reply:
Updated! I simplified and the parameterized test.

Reply:
looks a lot more clear and clean jere 👏